packages/contracts/contracts/universal/MultiSigWallet.sol

```diff
@@ -0,0 +1,587 @@
+    }
+
+    /**
+     * @notice Internal functions. Execute a confirmed transaction.
+     *
+     * @param _transactionId Transaction ID.
+     */
+    function _executeTransaction(uint256 _transactionId)
+        internal
+        ownerExists(msg.sender)
+        confirmed(_transactionId, msg.sender)
+        notExecuted(_transactionId)
+        nonReentrant
+    {
+        if (_isConfirmed(_transactionId)) {
+            Transaction storage txn = transactions[_transactionId];
+            txn.executed = true;
+            bool success = SafeCall.call(txn.destination, gasleft(), txn.value, txn.data);
+            require(success, "call transaction failed");
+            emit Execution(_transactionId);
+        }
+    }
+
+    /**
+     * @notice Allows anyone to execute a confirmed transaction.
+     *
+     * @param _transactionId Transaction ID.
+     */
+    function executeTransaction(uint256 _transactionId)
+        external
+        override
+    {
+        _executeTransaction(_transactionId);
+    }
+
+    /**
+     * @notice Internal functions. Returns the confirmation status of a transaction.
+     *
+     * @param _transactionId Transaction ID.
+     *
+     * @return Confirmation status.
+     */
+    function _isConfirmed(uint256 _transactionId)
+        internal
+        view
+        returns (bool)
+    {
+        uint256 count = 0;
+        mapping (address => bool) storage confirmation = confirmations[_transactionId];
+        for (uint256 i = 0; i < owners.length;) {
+            if (confirmation[owners[i]])
+                count += 1;
+            if (count == numConfirmationsRequired)
+                return true;
+
+            unchecked {
+                ++i;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * @notice Returns the confirmation status of a transaction.
+     *
+     * @param _transactionId Transaction ID.
+     *
+     *  @return Confirmation status.
+     */
+    function isConfirmed(uint256 _transactionId)
+        external
+        override
+        view
+        returns (bool)
+    {
+        return _isConfirmed(_transactionId);
+    }
+
+    /**
+     * @notice Adds a new transaction to the transaction mapping, if transaction does not exist yet.
+     *
+     * @param _destination Transaction target address.
+     * @param _value       Transaction ether value.
+     * @param _data        Transaction data payload.
+     *
+     * @return transactionId Returns transaction ID.
+     */
+    function addTransaction(address _destination, uint256 _value, bytes memory _data)
+        internal
+        notNull(_destination)
+        returns (uint256 transactionId)
+    {
+        transactionId = transactionCount;
+        transactions[transactionId] = Transaction({
+            destination: _destination,
+            value: _value,
+            data: _data,
+            executed: false
+        });
+        transactionCount += 1;
+        emit Submission(transactionId);
```

Comment:
```suggestion
        emit Submission(++transactionId);
```

Reply:
Applied.

Reply:
Oh.. you're right! yeah we can't reduce like that :(